nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/test/java/org/nuxeo/ecm/platform/audit/TestSimpleTemplateBasedRootAudit.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.platform.audit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.PathRef;
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.ACL;
+import org.nuxeo.ecm.core.api.security.ACP;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.query.api.PageProvider;
+import org.nuxeo.ecm.platform.query.api.PageProviderService;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/**
+ * The root init acl which happens at
+ * {@link org.nuxeo.ecm.platform.content.template.factories.SimpleTemplateBasedRootFactory}. Should occur only once, at
+ * the beginning when we add the first children of the root document.
+ * <p>
+ * Deploying a new contribution for a given test method, will call the {@code SimpleTemplateBasedRootFactory} for a
+ * second time, but as the root documents already contains children, we should not init the acl of the root document.
+ * 
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(AuditFeature.class)
+public class TestSimpleTemplateBasedRootAudit {
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected PageProviderService pps;
+
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.audit.tests:test-add-content-template-contrib.xml")
+    public void shouldInitRootAclOnlyOnce() {
+        DocumentModel root = session.getDocument(new PathRef("/"));
+        @SuppressWarnings("unchecked")
+        PageProvider<LogEntry> pp = (PageProvider<LogEntry>) pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", null, 3l,
+                0l, new HashMap<String, Serializable>(), root);
```

Comment:
In the future please use `L` instead of `l` for readability. `l` is very easily confused with `1`.